### PR TITLE
Implemented CRUD API Routes

### PR DESCRIPTION
**Pull request by PlasmaIntec**
Please review and merge @elizabethcan @pon-y

### CODE REVIEW

File: server/README.md [hunk 1]
```diff
@@ -98,7 +98,7 @@
 
 * **URL**
 
-  - `/api/details/`
+  - `/api/details`
```
Comment (elizabethcan) [inline server/README.md:101]: you need to add documentation about what parameters are required when hitting this api

Comment (elizabethcan) [reply server/README.md:101]: you also need to explain the Method and what the route does

File: server/README.md [hunk 2]
```diff
@@ -132,7 +132,7 @@
 
 * **URL**
 
-  - `/api/photos/:propertyId/:url`
+  - `/api/photos/:propertyId`
```
Comment (elizabethcan) [inline server/README.md:135]: see above

File: server/index.js [hunk 3]
```diff
@@ -52,9 +52,14 @@ app.get('/api/full/photos/:propertyId', (req, res) => {
...
-var { getRoute, postRoute, putRoute, deleteRoute} = require('./controller');
+var { postRoute, putRoute, deleteRoute } = require('./controller');
+var parser = require('body-parser');
 
-app.get('/api/')
+app.post('/api/basicdetails/', parser.json(), postRoute);
+
+app.put('/api/basicdetails/:propertyId', parser.json(), putRoute);
+
+app.delete('/api/basicdetails/:propertyId', deleteRoute);
```
Comment (elizabethcan) [inline server/index.js:62]: you might want to refactor these to conform to REST, i.e. something like `/properties/:propertyId/photos` -> this route should be descriptive of what information you are addressing, so, you're going through the route of properties, selecting a specific property, then getting all photos from that property. I know that you have legacy code to work with so it's up to you whether you want to address this in the future or not

**Review by elizabethcan — CHANGES_REQUESTED** [review]
Only blocking comment is updating the documentation for your API routes

File: server/index.js (hunk 3 above)
Comment (PlasmaIntec) [reply server/index.js:62]: @elizabethcan will address this after writing seed generation script

**Pull request merged**